app/controllers/staff/training_sessions_controller.rb

```diff
@@ -0,0 +1,91 @@
+class Staff::TrainingSessionsController < StaffAreaController
+  before_action :current_training_session, only: [:show, :edit, :update, :certify_trainees, :delete_training_session]
+  before_action :changed_params, only: [:update]
+
+  layout 'staff_area'
+
+  def index
+
+  end
+
+  def new
+    @new_training_session = TrainingSession.new
+  end
+
+  def create
+    @training_session = TrainingSession.new(training_session_params)
+    if params['training_session_users'].present?
+      JSON.parse(params['training_session_users']).each do |user|
+        @training_session.users << User.find_by(username: user)
+      end
+    end
+    if @training_session.save
+      redirect_to "#{staff_training_session_path(@training_session.id)}"
+    else
+      redirect_to :back
+      flash[:alert] = "Something went wrong. Please try again."
+    end
+  end
+
+  def update
+    #binding.pry
+    @current_training_session.update(changed_params)
+    if params['users'].present?
+      JSON.parse(params['users']).each do |new_user|
```

Comment:
If you pass the parameters in the right format you shouldn't need this JSON parsing. Something like [this stack overflow](https://stackoverflow.com/a/23066966)